File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/utils/time/NoneSpecifiedTimeFormat.java
package com.mapbox.services.android.navigation.v5.utils.time;


import java.util.Calendar;
import java.util.Locale;

import static com.mapbox.services.android.navigation.v5.utils.time.TwelveHoursTimeFormat.TWELVE_HOURS_FORMAT;
import static com.mapbox.services.android.navigation.v5.utils.time.TwentyFourHoursTimeFormat.TWENTY_FOUR_HOURS_FORMAT;

class NoneSpecifiedTimeFormat implements TimeFormatResolver {
  private final boolean isDeviceTwentyFourHourFormat;

  NoneSpecifiedTimeFormat(boolean isDeviceTwentyFourHourFormat) {
    this.isDeviceTwentyFourHourFormat = isDeviceTwentyFourHourFormat;
  }

  @Override
  public void nextChain(TimeFormatResolver chain) {
  }

  @Override
  public String obtainTimeFormatted(int type, Calendar time) {
    if (isDeviceTwentyFourHourFormat) {
      return String.format(Locale.getDefault(), TWENTY_FOUR_HOURS_FORMAT, time, time);
    } else {
      return String.format(Locale.getDefault(), TWELVE_HOURS_FORMAT, time, time, time);
    }
  }
}

File: libnavigation-model/src/main/java/com/mapbox/navigation/model/formatter/time/NoneSpecifiedTimeFormat.kt
package com.mapbox.services.android.navigation.v5.utils.time

import java.util.Calendar
import java.util.Locale

internal class NoneSpecifiedTimeFormat(
    private val isDeviceTwentyFourHourFormat: Boolean
) : TimeFormatResolver {

    override fun nextChain(chain: TimeFormatResolver) {}

    override fun obtainTimeFormatted(type: Int, time: Calendar): String {
        return if (isDeviceTwentyFourHourFormat) {
            String.format(
                Locale.getDefault(),
                TwentyFourHoursTimeFormat.TWENTY_FOUR_HOURS_FORMAT,
                time,
                time
            )
        } else {
            String.format(
                Locale.getDefault(),
                TwelveHoursTimeFormat.TWELVE_HOURS_FORMAT,
                time,
                time,
                time
            )
        }
    }
}
